### DISCUSSION TITLE
Return (Enter) key doesn’t work

### DISCUSSION THREAD

**Discussion by ZaharBozhok** [coder/code-server#5221] [Q&A]
Env:
- iPad Pro iPadOS 15.5
- code-server 4.4.0 on Ubuntu 20.04 
- used in Safari’s PWA
- no https

After some time Enter key doesn’t work in PWA. I connect to code-server using plain IP in VPN network.
I tried to close PWA and start again - enter doesn’t work.
Restarted server - also doesn’t help.
Closing PWA & Safari - doesn’t help.

iPad reboot only helps. Seems like iPad’s issue but enter works everywhere except the PWA.
1. Did somebody faced this issue and how it was resolved?
2. Maybe I can enable logs? If yes, then which and where?
([Link to the video](https://1drv.ms/v/s!AvXOm4PQKSxinVjJaEp3aJXM-K3n))

**Comment by jsjoeio** [comment]
I have had this happen as well. Can you reproduce it on vscode.dev on iPad? Wondering if it's upstream too.

**Reply by ZaharBozhok** [reply]
Okay, I will check. Did you reboot or fixed somehow in other way?

**Reply by jsjoeio** [reply]
Refreshing the PWA with CMD + R did it for me, but it was really annoying. I just haven't had time to debug it yet.

**Reply by ZaharBozhok** [reply]
Okay, I will check)

**Reply by jsjoeio** [reply]
Most important thing is seeing if it reproduces on vscode.dev, which I'm hopeful it does.

**Reply by ZaharBozhok** [reply]
I don’t know whether it reproduces in vscode.dev, but it reproduces in github codespaces

**Reply by ZaharBozhok** [reply]
And I don’t know why, but reloading the page doesn’t help. But helps unplugging and plugging back magic keyboard :)

**Reply by jsjoeio** [reply]
> but it reproduces in github codespaces

AHHHH! Nice find!!! Can you file an issue upstream and link it back here?

**Reply by ZaharBozhok** [reply]
[Link to the issue in github/vscode](https://github.com/microsoft/vscode/issues/151626)

**Reply by ZaharBozhok** [reply]
Also something weird happens to the Ctrl+` shortcut, it works randomly, no correlation with enter key. Mentioned in the upstream ticket

**Comment by mirekphd** [comment]
I think I know one of the reasons for this apparent bug. 

It happens when you have several files opened in code-server tabs and the Search / Find box had been populated with a search string that was found in one of these tabs, but then we switched to another tab and now the search string can no longer be found, and yet the Search / Find box still has the focus ... for the Enter key alone (any other key, including Ctrl+Enter combination, works normally). A red warning "No results" is displayed each time when we switch to the tab when the problem will occur. In other tabs (without this problem and this warning message), pressing Enter will move selection to consecutive occurrences of the search string.

**Reply by calvinvette** [reply]
This just happened to me (4.95/1.95), and I had just done a search and replace. 

I've tried tapping in and out of terminal, search, editor windows; using an external keyboard doesn't work; I've hit the Escape key; I've logged out and back in; crashed the app on the iPad and re-opened.

It still won't release the focus for the Return and other keys. Other keys like arrows and tab, and possibly Escape, also seem affected.

Is there another way to work around this easily?

**Reply by techlingyi** [reply]
@calvinvette Very late reply but for me, it fixes itself when I go back to the tab I did the search on, open the search box again with Ctrl+F, delete any words still in the search box, then hit the X/close button on the search box. (Works perfectly until I can't remember which tab I did the search on ^^;)

**Comment by HugoMcPhee** [comment]
Tapping in and out of the terminal fixes it for me
